src/plugins/AutoRefresh.ts

```diff
@@ -20,44 +19,30 @@ class AutoRefresh extends GenericPlugin {
   }
 
   public beforeClear() {
-    this.stopMonitoringViewability();
+    ScrollMonitor.unsubscribe(this.ad.el.id);
     dispatchEvent(this.ad.id, LOG_LEVELS.INFO, 'AutoRefresh Plugin', 'Ad viewability monitor has been removed.');
   }
 
   public beforeDestroy() {
-    this.stopMonitoringViewability();
+    ScrollMonitor.unsubscribe(this.ad.el.id);
     dispatchEvent(this.ad.id, LOG_LEVELS.INFO, 'AutoRefresh Plugin', 'Ad viewability monitor has been removed.');
   }
 
   private startMonitoringViewability(): void {
-    if (this.watcher) {
-      return;
-    }
-
-    const { container } = this.ad;
-
-    this.watcher = scrollMonitor.create(container);
+    const { container, configuration: { offset = 0 }, el } = this.ad;
+    ScrollMonitor.subscribe(
```

Comment:
Because the scroll monitor is assigned on a per ad basis, each scroll monitor ad has all of the attributes for that ad already. Even if another plugin subscribes the ad to the scroll monitor, it will have all of the same information. I don't believe there is a need to overwrite configuration options ever as long as when the ad is subscribed all the configuration is passed in.

Reply:
The `watcher` that was used by the plugin is no longer relevant with this new implementation.
If you foresee an issue that i am missing, which is totally possible, let's find an example so i can wrap my head around it.